### PR TITLE
Docs: Add migration docs for react-router v6 in plugins

### LINKED ISSUE #69059: docs: add guidance on React Router 6 upgrade
At minimum a new section under here https://grafana.com/docs/grafana/next/developers/plugins/migration-guide/v9.x-v10.x/ would be good to help plugin developers understand their options and how this support has been enabled.

Could also consider it's own section i.e. `/migration-guide/react-router-5-6/` and linking from the former.
```[tasklist]
### Tasks
- [ ] https://github.com/grafana/plugin-tools/pull/746
- [ ] https://github.com/grafana/plugin-tools/pull/707
- [ ] https://github.com/grafana/plugin-tools/pull/710
- [ ] https://github.com/grafana/scenes/pull/570
- [ ] https://github.com/grafana/grafana-plugins-platform-team/issues/81
- [ ] https://github.com/grafana/plugin-tools/pull/718
- [ ] https://github.com/grafana/grafana-plugin-examples/pull/245
- [ ] https://github.com/grafana/plugin-tools/pull/756
- [ ] https://github.com/grafana/grafana/issues/81988
- [x] Create an issue in `grafana/scenes` to update to react-router v6: https://github.com/grafana/scenes/issues/608
- [x] Add a post about the migration to the community forum (https://community.grafana.com/t/migrating-app-plugins-to-use-react-router-v6/115410)
- [x] Create a grouping issue for what needs to be done in core Grafana to migrate it to react-router v6 (https://github.com/grafana/grafana/issues/83666)
- [x] Create a grouping issue for tracking how internal app plugins update to react-router v6: https://github.com/grafana/grafana-plugins-platform-team/issues/85
```

### PR DESCRIPTION
**Pull request by leventebalogh**
Fixes: #69059 

### What changed?

Added a migration guide for updating react-router `v5` -> `v6` in plugins.

### Prerequisites

In order to have a migration doc backed by our plugin update tool and relevant examples, we have some prerequisites that we should do before updating the migration docs.

- [ ] **1.** Update the `@grafana/create-plugin update` command to be able to update both devDependencies and dependencies https://github.com/grafana/plugin-tools/pull/707
- [x] **2.** Add a feature flag to `@grafana/create-plugin` that allows to use the new react router (can be enabled for existing plugins and automatically received by new scaffolds) https://github.com/grafana/plugin-tools/pull/710
- [ ] **3.** Update the `grafana/grafana-plugin-examples` to use the latest router
- [ ] **4.** Update and release the migration doc

### CODE REVIEW

File: docs/sources/developers/plugins/migration-guide/v9.x-v10.x/_index.md [hunk 1]
```diff
@@ -34,3 +34,54 @@ Most code targeting 9.x will continue to work without any issues. An exception i
 When writing plugins that should run on 9.x, continue to use the Vector interfaces. In this case, when targeting versions 10+, you can now use simple arrays rather than wrapper classes.
 
 To make this transition seamless, we employed the Original JavaScript Sin™. That is, we[extended the native Array prototype](https://github.com/grafana/grafana/blob/v10.0.x/packages/grafana-data/src/types/vector.ts) with several Vector methods. We will atone and undo this in v11, when Vector interfaces and classes are removed.
+
+## Update to React Router v6
+
+Starting from Grafana 10 plugins can start updating to use the v6 version of `react-router`. Overall, `react-router` v6 aims to simplify route configuration and provide a more flexible and intuitive API for developers.
```
Comment (academo) [inline docs/sources/developers/plugins/migration-guide/v9.x-v10.x/_index.md:40]: Q. plugins `can` or `should` migrate? the line after implies is required.

Comment (leventebalogh) [reply docs/sources/developers/plugins/migration-guide/v9.x-v10.x/_index.md:40]: Yeah, I think it's good question. I would say it's rather a _should_, but I guess we would need to discuss it in the team.

File: docs/sources/developers/plugins/migration-guide/v9.x-v10.x/_index.md [hunk 2]
```diff
@@ -34,3 +34,54 @@ Most code targeting 9.x will continue to work without any issues. An exception i
...
+  <Routes>
+    {/* Until v5 */}
+    <Route exact path="/" element={<Home />} />
+
+    {/* From v6 */}
+    {/* (Routes are "exact" by default, you need to use the "*" to match sub-routes) */}
+    <Route path="/" element={<Home />} />
+  </Routes>
+);
+```
```
[the remark this answers is not in the document]
Comment (leventebalogh) [reply docs/sources/developers/plugins/migration-guide/v9.x-v10.x/_index.md:87]: Good point! Yes, currently after they update their plugin won't support anything lower than Grafana v10.

**Comment by sympatheticmoose** [comment]
Still some things to work out regarding implications on backwards compatibility for plugins and how it can be incorporated in create-plugin

**Review by josmperez — COMMENTED** [review]
This file has been moved from grafana repo to plugin-tools repo. If these changes are still needed, please create PR in other repo. Ask me if you have any questions.

**Comment by tolzhabayev** [comment]
@leventebalogh can we move this forward? It's currently hanging in the review status.

File: docs/sources/developers/plugins/migration-guide/v9.x-v10.x/_index.md [hunk 3]
```diff
@@ -34,3 +34,54 @@ Most code targeting 9.x will continue to work without any issues. An exception i
 When writing plugins that should run on 9.x, continue to use the Vector interfaces. In this case, when targeting versions 10+, you can now use simple arrays rather than wrapper classes.
 
 To make this transition seamless, we employed the Original JavaScript Sin™. That is, we[extended the native Array prototype](https://github.com/grafana/grafana/blob/v10.0.x/packages/grafana-data/src/types/vector.ts) with several Vector methods. We will atone and undo this in v11, when Vector interfaces and classes are removed.
```
Comment (josmperez) [inline docs/sources/developers/plugins/migration-guide/v9.x-v10.x/_index.md:36]: Is this an actual trademark? If not, and this is cheeky, I suggest putting the phrase "Original Javacript Sin" in scare quotes. If it is, ignore this comment.

File: docs/sources/developers/plugins/migration-guide/v9.x-v10.x/_index.md [hunk 4]
```diff
@@ -34,3 +34,54 @@ Most code targeting 9.x will continue to work without any issues. An exception i
...
+
+Starting from Grafana 10 plugins can start updating to use the v6 version of `react-router`. Overall, `react-router` v6 aims to simplify route configuration and provide a more flexible and intuitive API for developers.
+
+**We strongly encourage developers to update their plugins to use the v6 version `react-router` as soon as possible, as the v5 version is going to be deprecated in the future.**
+
+[Official React Router v5 to v6 migration guide](https://reactrouter.com/en/main/upgrading/v5)
+
+### Update using `@grafana/create-plugin`
+
+Please follow the steps below to start using react router v6 in your plugin.
```
Comment (josmperez) [inline docs/sources/developers/plugins/migration-guide/v9.x-v10.x/_index.md:48]: You can also replace "React router" with "`react-router`".

**Review by josmperez — APPROVED** [review]
Approved with a few nits to fix.

**Comment by leventebalogh** [comment]
Closing in favour of https://github.com/grafana/plugin-tools/pull/718.

**Pull request closed without merging**